v3/src/components/graph/adornments/lsrl/lsrl-adornment-component.tsx

```diff
@@ -122,7 +122,7 @@ export const LSRLAdornment = observer(function LSRLAdornment(props: IProps) {
         .style("top", `${top}px`)
 
       if (isFinished) {
-        const lines = model.lines.get(instanceKey)
+        const lines = dataConfig && model.getLines(xAttrId, yAttrId, cellKey, dataConfig)
```

Comment:
Given that this occurs multiple times in multiple other callbacks, you may want to pull it out into its own callback:
```
  const getLines = useCallback(() => {
    return dataConfig && model.getLines(xAttrId, yAttrId, cellKey, dataConfig)
  }, [cellKey, dataConfig, xAttrId, yAttrId])
```
which then shortens the dependency list for the other effects.